app/Http/Controllers/Service/RegistrationController.php

```diff
@@ -92,7 +92,49 @@ function ($child) {
             return redirect()->route('service.registration')->withErrors('Registration failed.');
         }
         // Or return the success
-        Log::info('Registration ' . $registration->id . ' stored by service user ' . Auth::id());
-        return redirect()->route('service.registration')->with('message', 'Registration saved.');
+        Log::info('Registration ' . $registration->id . ' created by service user ' . Auth::id());
+        return redirect()
+            ->route('service.edit_registration', ['id' => $registration->id ])
+            ->with('message', 'Registration created.');
+    }
+
+    public function update($id)
+    {
+        //
+    }
+
+    /**
+     * Show the Registration / Family edit form
+     *
+     * @param integer $id
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function edit($id)
+    {
+        // Get User and Centre;
+        // TODO: turn this into a masthead view composer on the app service provider.
+        $user = Auth::user();
+        $data = [
+            'user_name' => $user->name,
+            'centre_name' => ($user->centre) ? $user->centre->name : null,
+        ];
+
+        // Get the registration, with deep eager-loaded Family (with Children and Carers)
+        $registration = Registration::with([
+            'family' => function ($q) {
+                $q->with('children', 'carers');
+            }
+        ])->findOrFail($id);
+
+        return view('service.edit_registration', array_merge(
+            $data,
+            [
+                'registration' => $registration,
+                'family' => $registration->family,
+                'pri_carer' => $registration->family->carers->shift(),
+                'sec_carers' => $registration->family->carers,
```

Comment:
Does shift lop the first value off the array so that sec_carers will not contain the pri_carer?

Even though you access carers again through the family?

Reply:
Yes, once family is in memory, it's just an array and doesn't seem to refresh from database unless a saving event happens. You are correct, though, it may be better to copy the array into another and array_shift() that.